### ISSUE TITLE
Radiance RGBE (*.hdr) file format support

### ISSUE THREAD

**Issue by NicTanghe** [Beep6581/RawTherapee#6606]
Hello,

I have looked around everywhere, and the only current HDRI colour grading workflow
is to grade all the images separately before you merge them into an HDRI.

This is incredibly unwieldy. And doesn't allow you to grade HDRI`s for which you do not have the source images.

example [industrial_sunset_02_2k.hdr](https://dl.polyhaven.org/file/ph-assets/HDRIs/hdr/2k/industrial_sunset_02_2k.hdr)

**Comment by Entropy512** [comment]
Not sure if this would be solved by moving to libraw as I'm not sure if libraw supports importing the HDRI format.

A workaround would be to convert the HDRIs to float32 DNG.  RT definitely supports the latter.

**Comment by NicTanghe** [comment]
Hej thanks for the workaround I'll go test if that works.
Should be great if it does that, And raw therapy just does the conversion automatically in working space.

**Comment by Beep6581** [comment]
Similar: #1895

**Comment by Entropy512** [comment]
As a side note, a Python script I recently wrote to handle https://github.com/Beep6581/RawTherapee/issues/6612 also handles the provided Radiance RGBE image (since that is supported by imagecodecs by @cgohlke also).  I'll post that as a workaround soon.

**Comment by NicTanghe** [comment]
How would I use said python script ?

**Comment by NicTanghe** [comment]
can i just feed this an hdr and will it work ?
```python
#!/usr/bin/env python3

import imagecodecs
import tifffile
import sys
import os
import pyexiv2
import base64

#Base64 encoded linear (gamma=1.0) sRGB ICC profile generated by RT's ICC profile generator
#Anyone know of a decent way to create an ICC profile from Python?  Nearly everything I've seen is
#read-only
icc_profile = base64.b64decode('AAACsGxjbXMEMAAAbW50clJHQiBYWVogB+YACwAdAA4ALAAbYWNzcEFQUEwAAAAAAAAAAAAAAAAA\
AAAAAAAAAAAAAAAAAPbWAAEAAAAA0y1sY21zAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAAA\
AAAAAAAAAAAAAAAAAAAAAAAMZGVzYwAAARQAAAA8Y3BydAAAAVAAAABad3RwdAAAAawAAAAUY2hh\
ZAAAAcAAAAAsclhZWgAAAewAAAAUYlhZWgAAAgAAAAAUZ1hZWgAAAhQAAAAUclRSQwAAAigAAAAQ\
Z1RSQwAAAjgAAAAQYlRSQwAAAkgAAAAQY2hybQAAAlgAAAAkZG1uZAAAAnwAAAAybWx1YwAAAAAA\
AAABAAAADGVuVVMAAAAgAAAAHABzAFIARwBCAEwAaQBuAGUAYQByAF8AZwA9ADEALgAwbWx1YwAA\
AAAAAAABAAAADGVuVVMAAAA+AAAAHABDAG8AcAB5AHIAaQBnAGgAdAAgAFIAYQB3AFQAaABlAHIA\
YQBwAGUAZQAgADIAMAAyADEALAAgAEMAQwAwAABYWVogAAAAAAAA9tYAAQAAAADTLXNmMzIAAAAA\
AAEMQgAABd7///MlAAAHkwAA/ZD///uh///9ogAAA9wAAMBuWFlaIAAAAAAAAG+gAAA49QAAA5BY\
WVogAAAAAAAAJJ8AAA+EAAC2w1hZWiAAAAAAAABilwAAt4cAABjZcGFyYQAAAAAAAAAAAAEAAHBh\
cmEAAAAAAAAAAAABAABwYXJhAAAAAAAAAAAAAQAAY2hybQAAAAAAAwAAAACj1wAAVHsAAEzNAACZ\
mgAAJmYAAA9cbWx1YwAAAAAAAAABAAAADGVuVVMAAAAWAAAAHABSAGEAdwBUAGgAZQByAGEAcABl\
AGUAAA==')

if len(sys.argv) < 2:
    print("Need to provide filename!")
    exit(0)

hdrimage = imagecodecs.imread(sys.argv[1])

#Discard alpha channel
hdrimage = hdrimage[:,:,0:3]

filebase = os.path.splitext(sys.argv[1])[0]
tifname = filebase + '.tif'

with tifffile.TiffWriter(tifname) as tif:
    tif.write(hdrimage,
            photometric='rgb',
            compression='zlib',
            predictor=True)

with pyexiv2.Image(tifname) as exiv_image:
    exiv_image.modify_icc(icc_profile)
    ```

**Comment by NicTanghe** [comment]
Can anyone point me in the general direction where in the codebase files and images are loaded and converted ?

I really really mis HDRI / EXR support.

Grading hdri`s after merging them from different exposures instead of having to manually grade all exposure levels (which is super prone to error) would be soooooooo useful.

**Comment by Lawrence37** [comment]
@NicTanghe Have a look at the JXL import pull request (#6367). `rtengine/imageio.cc` seems like a good starting point.

**Comment by kmilos** [comment]
> I really really mis HDRI / EXR support.

If you end up working on EXR import, it might be worth considering whether you pull in the larger official OpenEXR library as dependency, or go with a smaller (but maybe not feature complete) [tinyexr](https://github.com/syoyo/tinyexr) one...

And I guess [EXR (at least 16-bit half float) is preferable to RGBE](http://www.anyhere.com/gward/hdrenc/).

**Comment by NicTanghe** [comment]
Hello @kmilos i don't exactly understand the statement  
"EXR (at least 16-bit half float) is preferable to RGBE"
Isn`t 16 bit float just a data format for the variables in RGBE ?

**Comment by kmilos** [comment]
> Isn`t 16 bit float just a data format for the variables in RGBE ?

No, RGBE uses a different representation (a single, shared exponent and fewer bits for the individual 3 mantissas), and as a consequence has poorer relative precision as explained in the article.

**Comment by NicTanghe** [comment]
How is "uses a different representation" different from another data format ?
(serious question)

**Comment by NicTanghe** [comment]
Thx for the lesson on nomenclature.
( i gave up quite fast at doing this)
